File: commands/h.js
let clc = require("cli-color");
let moment = require("moment");
require("moment-timezone");
require("moment-duration-format");
const RSSParser = require("rss-parser");

module.exports.run = async (client, message, args, cmdHook, roCMD, DBResult) => {

    let [cmdRegion, ...ignored] = args;

    if(DBResult.command === false && message.author.id !== config.ownerID) {
        client.misc.cmdHook(message.content, "fail", "PERM", message.author, message.guild, null);
        return message.channel.send({embed: {
            color: 0xc6373e,
            author: {
              name: client.user.username,
              icon_url: client.user.displayAvatarURL
            },
            title: "Error!",
            description: "This guild has disabled the use of `h`.",
            footer: {
                icon_url: message.author.displayAvatarURL,
                text: message.author.username
            }
        }});
    }

    if(cmdRegion === undefined){
        cmdRegion = DBResult.region
    }

    if(moment.tz.zone(`${cmdRegion}`)) {
        if(DBResult.adult === true){
            adultURL = "&adult=true";
        } else {
            adultURL = "";
        };

        var cmdURL = `https://www.checkiday.com/rss.php?tz=${cmdRegion}${adultURL}`;
        
        let items = "";
        let parser = new RSSParser();
        parser.parseURL(cmdURL, function(err, feed) {
            try {feed.items.forEach(item => items += `\n\n` + "• " + `**${item.title}**` + "")}
            catch (error) {
                client.misc.cmdHook(message.content, "fail", "ERR", message.author, message.guild, null);
                return message.channel.send({embed: {
                    color: 0xc6373e,
                    author: {
                        name: client.user.username,
                        icon_url: client.user.displayAvatarURL
                    },
                    title: "Error!",
                    description: "There was an error gathering the feed from [Checkiday](https://checkiday.com) (This is most likely their fault!)",
                    footer: {
                        icon_url: message.author.displayAvatarURL,
                        text: message.author.username
                    }
                }});
            }
            message.channel.send({embed: {
                color: 0x10525C,
                author: {
                    name: client.user.username,
                    icon_url: client.user.displayAvatarURL
                },
                title: `Today's Holidays in region ${cmdRegion}`,
                url: "https://checkiday.com",
                fields: [{
                    name: `${feed.pubDate}`.slice(0, 16),
                    value: items
                }],
                footer: {
                    icon_url: message.author.displayAvatarURL,
                    text: message.author.username
                }
            }});
        });
    } else {
        client.misc.cmdHook(message.content, "fail", "SYN", message.author, message.guild, null);
        return message.channel.send({embed: {
            color: 0xc6373e,
            author: {
              name: client.user.username,
              icon_url: client.user.displayAvatarURL
            },
            title: "Error!",
            description: `Not a valid tz/zoneinfo database region (e.g. \`${client.misc.getDefaultRegion(message.guild)}\`). See list of all valid regions [here](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones).`,
            footer: {
                icon_url: message.author.displayAvatarURL,
                text: message.author.username
            }
        }});
    }
    client.misc.cmdHook(message.content, "succ", null, message.author, message.guild, null);
};

module.exports.help = {
    name: "h"
}

File: commands/h.go
package commands

import (
	"time"

	"github.com/barkloaf/HolidayBot/config"
	"github.com/barkloaf/HolidayBot/misc"
	"github.com/bwmarrin/discordgo"
)

func h(p Params) bool {
	tz := p.DBResult.Region

	_, err := time.LoadLocation(p.Args[1])
	if err == nil && p.Args[1] != "" {
		tz = p.Args[1]
	}

	field, err := misc.Feed(tz, p.DBResult.Adult)
	if err != nil {
		Errors(p.Client, p.Message, p.Guild, "FEED", H)
		return false
	}

	embed := &discordgo.MessageEmbed{
		Author: &discordgo.MessageEmbedAuthor{
			Name:    p.Client.State.User.Username,
			IconURL: p.Client.State.User.AvatarURL(""),
		},
		Color:  config.Config.UseColor,
		Title:  "Today's Holidays (" + tz + "):",
		URL:    "https://checkiday.com",
		Fields: field,
		Footer: &discordgo.MessageEmbedFooter{
			Text:    p.Message.Author.Username,
			IconURL: p.Message.Author.AvatarURL(""),
		},
	}

	p.Client.ChannelMessageSendEmbed(p.Message.ChannelID, embed)

	return true
}
